vector_search/tasks.py

```diff
@@ -0,0 +1,143 @@
+import logging
+
+import celery
+from celery.exceptions import Ignore
+from django.conf import settings
+
+from learning_resources.etl.constants import RESOURCE_FILE_ETL_SOURCES
+from learning_resources.models import (
+    ContentFile,
+    Course,
+    LearningResource,
+)
+from learning_resources.utils import load_course_blocklist
+from learning_resources_search.constants import (
+    CONTENT_FILE_TYPE,
+    COURSE_TYPE,
+    LEARNING_PATH_TYPE,
+    PODCAST_EPISODE_TYPE,
+    PODCAST_TYPE,
+    PROGRAM_TYPE,
+    SEARCH_CONN_EXCEPTIONS,
+    VIDEO_PLAYLIST_TYPE,
+    VIDEO_TYPE,
+)
+from learning_resources_search.exceptions import RetryError
+from learning_resources_search.tasks import wrap_retry_exception
+from main.celery import app
+from main.utils import (
+    chunks,
+)
+from vector_search.utils import embed_learning_resources
+
+log = logging.getLogger(__name__)
+
+
+@app.task(
+    acks_late=True,
+    reject_on_worker_lost=True,
+    autoretry_for=(RetryError,),
+    retry_backoff=True,
+    rate_limit="600/m",
+)
+def generate_embeddings(ids, resource_type):
+    """
+    Generate learning resource embeddings and index in Qdrant
+
+    Args:
+        ids(list of int): List of resource id's
+        resource_type (string): resource_type value for the learning resource objects
+
+    """
+    try:
+        with wrap_retry_exception(*SEARCH_CONN_EXCEPTIONS):
+            embed_learning_resources(ids, resource_type)
+    except (RetryError, Ignore):
+        raise
+    except SystemExit as err:
+        raise RetryError(SystemExit.__name__) from err
+    except:  # noqa: E722
+        error = "generate_embeddings threw an error"
+        log.exception(error)
+        return error
+
+
+@app.task(bind=True)
+def start_embed_resources(self, indexes, skip_content_files):
```

Comment:
Add a docstring with `Args` and `Returns` sections (and/or add argument/return types above)